io/plugins/eu.esdihumboldt.hale.io.xtraserver.test/src/eu/esdihumboldt/hale/io/xtraserver/writer/XtraServerMappingFileWriterTest.java

```diff
@@ -0,0 +1,247 @@
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     Data Harmonisation Panel <http://www.dhpanel.eu>
+ *     interactive instruments GmbH <http://www.interactive-instruments.de>
+ */
+
+package eu.esdihumboldt.hale.io.xtraserver.writer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import de.fhg.igd.slf4jplus.ALogger;
+import de.fhg.igd.slf4jplus.ALoggerFactory;
+import eu.esdihumboldt.hale.common.align.io.AlignmentIO;
+import eu.esdihumboldt.hale.common.align.io.AlignmentReader;
+import eu.esdihumboldt.hale.common.align.io.impl.AbstractAlignmentWriter;
+import eu.esdihumboldt.hale.common.align.io.impl.JaxbAlignmentReader;
+import eu.esdihumboldt.hale.common.align.model.Alignment;
+import eu.esdihumboldt.hale.common.core.HalePlatform;
+import eu.esdihumboldt.hale.common.core.io.IOProvider;
+import eu.esdihumboldt.hale.common.core.io.IOProviderConfigurationException;
+import eu.esdihumboldt.hale.common.core.io.PathUpdate;
+import eu.esdihumboldt.hale.common.core.io.extension.IOProviderDescriptor;
+import eu.esdihumboldt.hale.common.core.io.extension.IOProviderExtension;
+import eu.esdihumboldt.hale.common.core.io.impl.LogProgressIndicator;
+import eu.esdihumboldt.hale.common.core.io.project.ProjectInfo;
+import eu.esdihumboldt.hale.common.core.io.project.ProjectInfoAware;
+import eu.esdihumboldt.hale.common.core.io.project.impl.ArchiveProjectReader;
+import eu.esdihumboldt.hale.common.core.io.project.model.IOConfiguration;
+import eu.esdihumboldt.hale.common.core.io.project.model.Project;
+import eu.esdihumboldt.hale.common.core.io.project.model.ProjectFileInfo;
+import eu.esdihumboldt.hale.common.core.io.report.IOReport;
+import eu.esdihumboldt.hale.common.core.io.supplier.DefaultInputSupplier;
+import eu.esdihumboldt.hale.common.core.io.supplier.FileIOSupplier;
+import eu.esdihumboldt.hale.common.instance.io.InstanceReader;
+import eu.esdihumboldt.hale.common.schema.io.SchemaIO;
+import eu.esdihumboldt.hale.common.schema.io.SchemaReader;
+import eu.esdihumboldt.hale.common.schema.model.Schema;
+import eu.esdihumboldt.hale.common.schema.model.impl.DefaultSchemaSpace;
+import eu.esdihumboldt.hale.io.appschema.AppSchemaIO;
+
+@SuppressWarnings("javadoc")
+public class XtraServerMappingFileWriterTest {
+
+	private static final ALogger log = ALoggerFactory
+			.getLogger(XtraServerMappingFileWriterTest.class);
+
+	private static final String PROJECT_LOCATION = "/data/simpledemo.halez";
```

Comment:
Generally it would be preferable to have a test project as `.halex`, as that allows to track changes to the project and alignment - but that is just a suggestion.

Reply:
I would prefer to leave this untouched for now. We plan to rework and extend the tests in an upcoming project, I would follow your suggestion then.